src/examples/caninterrupt/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"device/sam"
+	"fmt"
+	"machine"
+	"time"
+)
+
+type canMsg struct {
+	ch   byte
+	id   uint32
+	dlc  byte
+	data []byte
+}
+
+func main() {
+	ch := make(chan canMsg, 10)
+	go func() {
+		for {
+			select {
+			case m := <-ch:
+				fmt.Printf("%d %03X %X ", m.ch, m.id, m.dlc)
+				for _, d := range m.data {
+					fmt.Printf("%02X ", d)
+				}
+				fmt.Printf("\r\n")
+			}
+
+		}
+	}()
+
+	can1 := machine.CAN{Bus: sam.CAN1}
+	can1.Configure(machine.CANConfig{
+		TransferRate:   machine.CANTransferRate500kbps,
+		TransferRateFD: machine.CANTransferRate1000kbps,
+		Rx:             machine.CAN1_RX,
+		Tx:             machine.CAN1_TX,
+	})
+	can1.SetInterrupt(sam.CAN_IE_RF0NE, func(machine.CAN) {
```

Comment:
What does `sam.CAN_IE_RF0NE` mean?

Also, I think you'll want to pass a pointer, not a value:

```suggestion
	can1.SetInterrupt(sam.CAN_IE_RF0NE, func(*machine.CAN) {
```

Reply:
Changed.

Reply:
`sam.CAN_IE_RF0NE` is  `Rx FIFO 0 New Message Interrupt Enable` register.
This interrupt is generated every time a CAN frame is received.

Reply:
The RF0NE is described in the following page 1141.

https://cdn-learn.adafruit.com/assets/assets/000/097/530/original/SAM_D5xE5x_Family_Data_Sheet_DS60001507F.pdf?1607462315